helpers/update-content.js

```diff
@@ -1,31 +1,171 @@
+            return !ruleName.includes('/');
+        });
+
+        /*
+         * Rules with id: package-name/rule-name will be
+         * considered part of a package with multiple rules
+         */
+        const multiRules = includedRules.filter((ruleName) => {
+            return ruleName.includes('/');
+        });
+
+        /*
+         * Group rules by package name.
+         */
+        const packagesName = _.groupBy(multiRules, (ruleName) => {
+            return ruleName.split('/')[0];
+        });
+
+        const rules = _.map(singleRules, processRule);
+
+        const multiRulesProcessed = _.reduce(packagesName, (multi, values, key) => {
+            // Add an item with the link to the package with multiple rules itself.
+            const partial = processRule(key, true);
+
+            multi.push(partial);
+
+            // Add an item for each individual rule for a package with multiple rules.
+            return multi.concat(_.map(values, processRule));
+        }, []);
+
+        const processedCategory = {
+            link: `/docs/user-guide/rules/${category}/`,
+            name: category,
+            rules: rules.concat(multiRulesProcessed)
+        };
+
+        allCategories.push(processedCategory);
+
+        return allCategories;
+    }, []);
+
+    return { categories: processedCategories };
+};
 
 config.fatal = true;
 
 exec(`git clone ${CLONE_URL}  "${TMP_DIR}"`);
 
-rm('-rf', `${SOURCE_DIR}/docs/contributor-guide`);
-rm('-rf', `${SOURCE_DIR}/docs/user-guide`);
-rm('-rf', `${SOURCE_DIR}/about`);
+rm('-rf', `${DEST_DIR}/docs/contributor-guide`);
+rm('-rf', `${DEST_DIR}/docs/user-guide`);
+rm('-rf', `${DEST_DIR}/about`);
+
+cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/contributor-guide`, `${DEST_DIR}/docs/contributor-guide`);
+cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/user-guide`, `${DEST_DIR}/docs/user-guide`);
+cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/about`, `${DEST_DIR}`);
+cp(`${PACKAGES_TMP_DIR}/sonarwhal/CHANGELOG.md`, `${DEST_DIR}/about`);
 
-cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/contributor-guide`, `${SOURCE_DIR}/docs/contributor-guide`);
-cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/user-guide`, `${SOURCE_DIR}/docs/user-guide`);
-cp('-R', `${PACKAGES_TMP_DIR}/sonarwhal/docs/about`, `${SOURCE_DIR}`);
-cp(`${PACKAGES_TMP_DIR}/sonarwhal/CHANGELOG.md`, `${SOURCE_DIR}/about`);
+const ruleDocs = ls('-R', `${PACKAGES_TMP_DIR}/rule-*/{README.md,/docs/*.md}`);
+const rules = ls('-R', `${PACKAGES_TMP_DIR}/rule-*/src/!(index).ts`);
 
-const rules = ls('-R', `${PACKAGES_TMP_DIR}/rule-*/README.md`);
+// Create folder if not exist before writing file.
+// Reference: https://stackoverflow.com/a/16317628
+const safeWriteFile = (dir, content) => {
+    const folderPath = getDirname(dir);
 
+    return mkdirp(folderPath, (err) => {
+        if (err) {
+            console.error(`Error creating path ${folderPath}: ${err.message}.`);
+        }
+
+        fs.writeFile(dir, content);
+    });
```

Comment:
I think there are some possible issues with this function. It is called synchronously in line 168 but it is definitely not. A quick fix would be to use [`mkdirp.sync`](https://www.npmjs.com/package/mkdirp#mkdirpsyncdir-opts) and `fs.writeFileSync`.

Reply:
Or use [`shelljs` `mkdir` with the option `-p`](https://github.com/shelljs/shelljs#mkdiroptions-dir_array) and remove the `mkdirp` dependency (although I think I'd rather use `mkdirp.sync`)